diff --git a/index.php b/index.php
index 5b074a6..674cee0 100755
--- a/index.php
+++ b/index.php
@@ -1,6 +1,23 @@
 <?php
+    function isHttps()
+    {
+        if (array_key_exists("HTTPS", $_SERVER) && 'on' === $_SERVER["HTTPS"]) {
+            return true;
+        }
+        if (array_key_exists("SERVER_PORT", $_SERVER) && 443 === (int)$_SERVER["SERVER_PORT"]) {
+            return true;
+        }
+        if (array_key_exists("HTTP_X_FORWARDED_SSL", $_SERVER) && 'on' === $_SERVER["HTTP_X_FORWARDED_SSL"]) {
+            return true;
+        }
+        if (array_key_exists("HTTP_X_FORWARDED_PROTO", $_SERVER) && 'https' === $_SERVER["HTTP_X_FORWARDED_PROTO"]) {
+            return true;
+        }
+        return false;
+    }
     $path = rtrim(str_replace('\\','/',dirname($_SERVER['PHP_SELF'])), '/');
-    $url = ((isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https://' : 'http://' ) . $_SERVER['HTTP_HOST'] . $path;
+    $url_host = getenv("HOST", false) !== false ? getenv("HOST", false) : $_SERVER['HTTP_HOST'];
+    $url = ( isHttps() ? 'https://' : 'http://' ) . $url_host . $path;
 ?>
 <!doctype html>
 <html class="no-js" lang="">
